docs/performance_benchmark.md

```diff
@@ -0,0 +1,289 @@
+Number of runs: 5
+Number of time terminated: 0
+Time passed (seconds): 3.0087939262390138 ± 0.0001147521277690166
+cpu min (%): 0.0 ± 0.0
+cpu max (%): 11.0 ± 2.23606797749979
+cpu mean (%): 6.2 ± 0.18257418583505522
+mem min (kb): 54.0265625 ± 0.11180339887498948
+mem max (kb): 54.0265625 ± 0.11180339887498948
+mem mean (kb): 54.0265625 ± 0.11180339887498948
+== Report created 2020-04-27 15:38:32.947308 ==
+Arguments are `(3, 0.001)`
+Number of runs: 5
+Number of time terminated: 0
+Time passed (seconds): 3.014109659194946 ± 0.0004416575764579524
+cpu min (%): 0.0 ± 0.0
+cpu max (%): 8.0 ± 2.7386127875258306
+cpu mean (%): 1.9986666666666666 ± 0.002981423969999689
+mem min (kb): 53.9890625 ± 0.10431954926750306
+mem max (kb): 53.9890625 ± 0.10431954926750306
+mem mean (kb): 53.9890625 ± 0.10431954926750306
+== Report created 2020-04-27 15:38:48.067511 ==
+Arguments are `(3, 0.01)`
+Number of runs: 5
+Number of time terminated: 0
+Time passed (seconds): 3.0181806087493896 ± 0.0022409499756841883
+cpu min (%): 0.0 ± 0.0
+cpu max (%): 1.0 ± 2.23606797749979
+cpu mean (%): 0.06666666666666667 ± 0.14907119849998599
+mem min (kb): 53.9078125 ± 0.11487297672320501
+mem max (kb): 53.9078125 ± 0.11487297672320501
+mem mean (kb): 53.9078125 ± 0.11487297672320501
+
+```
+
+Chart is drawn for argument 1: sleep:
+<center>![Char over argument 1 - sleep value](assets/benchmark_chart.png)</center>
+
+The most interesting part is CPU usage, as you can see  cPU usage decreases with increasing value of idle sleep.
+Memory usage and execution time can slightly differ per case execution.
+
+
+## Requirements to tested function
+* The first function's argument has to be `benchmark: BenchmarkControl` it passed by default by framework.
+* All arguments except the fist one have to set default values.
+* Function docstring is a must, it used for help information.
+* `benchmark.start()` has to be called once in the function body to start measurement. timeout is counted from this point!
+* All prepare part in the function that should not be measured has to be placed before `benchmark.start()`
+* Code to be measured has to go after `benchmark.start()`
+* Try to avoid infinitive loops and suppose test should exit after a while.
+
+
+## Execution options
+* To pass arguments set just provide it comma separated like `10,0.1`
+* To pass several arguremnt sets just separate it by space `10,0.1 20,0.2`
+* `--timeout FLOAT` is test execution timeout in seconds, it test takes more time, it will be terminated.
+* `--period FLOAT` is measurement interval in seconds, how often to make CPU and RAM usage measurements.
+* `-N, --num-executions INTEGER` - how many time run the same argument set to make result more accurate.
+* `-P, --plot INTEGER` -  Draw a chat using, using values of argument specified as values for axis X. argument positions started with 0, argument benchmark does not counted. for example `-P 0` will use `run_time` values, `-P 1` will use `sleep` values.
```

Comment:
```suggestion
## Requirements for tested function

* The first function's argument has to be `benchmark: BenchmarkControl` which is passed by default by the framework.
* All arguments except the fist one have to set default values.
* Function doc string is required, it used for help information.
* `benchmark.start()` has to be called once in the function body to start measurement. The timeout is counted from this point!
* All the "prepare part" in the function that should not be measured has to be placed before `benchmark.start()`
* Code to be measured has to go after `benchmark.start()`
* Try to avoid infinitive loops and assume the test should exit after a while.


## Execution options

* To pass an arguments set just provide it as a comma separated string like `10,0.1`
* To pass several argument sets just separate them by whitespace `10,0.1 20,0.2`
* `--timeout FLOAT` is test execution timeout in seconds. If the test takes more time, it will be terminated.
* `--period FLOAT` is measurement interval in seconds, how often to make CPU and RAM usage measurements.
* `-N, --num-executions INTEGER` - how many time to run the same argument set to make result more accurate.
* `-P, --plot INTEGER` -  Draw a chart using, using values of argument specified as values for axis X. argument positions started with 0, argument benchmark does not counted. for example `-P 0` will use `run_time` values, `-P 1` will use `sleep` values.
```